vizro-core/src/vizro/managers/_data_manager.py

```diff
@@ -1,126 +1,282 @@
+        >>> data_manager["dynamic_data"] = dynamic_data
+        >>> data_manager["dynamic_data"].timeout = 5  # if you want to change the cache timeout to 5 seconds
+
+    Possibly in future, this will become a public class so you could directly do:
+        >>> data_manager["dynamic_data"] = DynamicData(dynamic_data, timeout=5)
+    But we'd need to make sure that name is not an argument in __init__ then.
+
+    At this point we might like to disable the behaviour so that data_manager setitem and getitem handle the same
+    object rather than doing an implicit conversion to _DynamicData.
+    """
+
+    def __init__(self, name: str, load_data: pd_DataFrameCallable):
+        self.__load_data: pd_DataFrameCallable = load_data
+        # name is needed for the cache key and should not be modified by users.
+        self._name = name
+        self.timeout: Optional[int] = None
+        # We might also want a self.cache_arguments dictionary in future that allows user to customise more than just
+        # timeout, but no rush to do this since other arguments are unlikely to be useful.
+
+    @memoize
+    def load(self) -> pd.DataFrame:
+        """Loads data.
+
+        In future this will probably take arguments that are passed through to __load_data in order to re-run the
+        loading function with different arguments. We might want to use CapturedCallable for self.__load_data then.
+        """
+        # Note you get the same "cache missed" message if NullCache is running or if data_manager._cache_has_app is
+        # False.
+        logger.debug("Cache miss; reloading data")
+        return self.__load_data()
+
+    def __repr__(self):
+        """Flask-caching uses repr to form the cache key, so this is very important to set correctly.
+
+        In particular, it must depend on something that uniquely labels the data source and is the same across all
+        workers: self._name. Using id(self), as in Python's default repr, only works in the case that gunicorn is
+        running with --preload: without preloading, the id of the same data source in different processes will be
+        different so the cache will not match up.
+
+        flask_caching make it possible to set a __cached_id__ attribute to handle this so that repr can be set
+        independently of cache key, but this doesn't seem to be well documented or work well, so it's better to rely
+        on __repr__.
+        """
+        # Note that using repr(self.__load_data) is not good since it depends on the id of self.__load_data and so
+        # would not be consistent across processes.
+        # lambda function has __qualname__. partial does not unless explicitly assigned a name but will not work with
+        # flask-caching anyway since flask_caching.utils.function_namespace would not be able to find a name for it.
+        return f"{self.__class__.__name__}({self._name!r}, {self.__load_data.__qualname__})"
+
+
+class _StaticData:
```

Comment:
Ahhh this is a great question and a previous implementation had exactly what you describe. I went back and forth between many options here but am confident the current solution is best.

There are a few reasons to do it this way:
* static data vs. dynamic data should have different API, e.g. you can't set `timeout` to anything other than 0 for a static dataset, and likely more differences in future. This is much harder uglier to implement if you just make static data a particular type of dynamic data
* having `SimpleCache` as default is problematic because it's not obvious that when you move to production you'd need to change that, even if you're not using any dynamic data. This is a very easy to trap to fall into (I ran into it myself many times by accident so decided it was too risky for real users)
* when we add the ability to manually reload the data through a button on the UI (coming soon) then there's as distinction between `timeout=0` dynamic data, which can be forcibly refreshed, vs. static data which can never be refreshed